src/couch/src/couch_att.erl

```diff
@@ -412,13 +412,21 @@ follow_from_json(Att, Props) ->
 
 inline_from_json(Att, Props) ->
     B64Data = couch_util:get_value(<<"data">>, Props),
-    Data = base64:decode(B64Data),
-    Length = size(Data),
-    RevPos = couch_util:get_value(<<"revpos">>, Props, 0),
-    store([
-        {data, Data}, {revpos, RevPos}, {disk_len, Length},
-        {att_len, Length}
-    ], Att).
+    try base64:decode(B64Data) of
+        Data ->
+            Length = size(Data),
+            RevPos = couch_util:get_value(<<"revpos">>, Props, 0),
+            store([
+                {data, Data}, {revpos, RevPos}, {disk_len, Length},
+                {att_len, Length}
+            ], Att)
+    catch
+        _:_ ->
```

Comment:
Wonder if we should be more conservative and explicitly match on the knows exception tags and error message

```
error:function_clause
 
error:{badmatch, _}
```
the `function_clause` happens when is an invalid input data type such as undefined or list.

`badmatch` happens if type is correct but data is not a valid B64 encoding.

But I guess it's not clear that there couldn't be other errors thrown so maybe catching all is ok there. Docs don't say what exactly should be thrown.

So up to you here. I am 50/50 on this one.

Reply:
I thought about it, but then decided to opt out. We don't particularly interested in _how_ `base64` can't decode our data, only in fact that it can't be decoded, so there are no merit to try and match all possible ways `base64` can break, if at the end we are raising the same invalid request exception.

Reply:
Yap got it, that makes sense